domains/client/block-preprocessor/src/inherents.rs

```diff
@@ -81,19 +172,32 @@ where
     CClient: ProvideRuntimeApi<CBlock> + HeaderBackend<CBlock>,
     CClient::Api: DomainsApi<CBlock, NumberFor<Block>, Block::Hash>,
 {
-    // TODO: we need to include the runtime upgrade
-    type InherentDataProviders = sp_timestamp::InherentDataProvider;
+    type InherentDataProviders = (
+        sp_timestamp::InherentDataProvider,
+        sp_executive::InherentDataProvider,
+    );
 
     async fn create_inherent_data_providers(
         &self,
         _parent: Block::Hash,
         _extra_args: (),
     ) -> Result<Self::InherentDataProviders, Box<dyn Error + Send + Sync>> {
-        let best_consensus_hash = self.consensus_client.info().best_hash;
+        let consensus_block_hash = self
+            .maybe_consensus_block_hash
+            .unwrap_or(self.consensus_client.info().best_hash);
```

Comment:
It is hard to see what the value of `maybe_consensus_block_hash` would be and when it will be `None`, plz add some comments here to help understand it.